### PR TITLE
Dev favorite opponent

### PR DESCRIPTION
**Pull request by dunlapww**
Add Favorite Opponent and Rival all tests passing so is spec harness

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -213,8 +189,8 @@ def games_by_team(team_id)
     end
   end
 
-  def get_opponent_id(game, team_id)
-    @games_manager.get_opponent_id(game, team_id)
+  def get_opponent_id(gameid, teamid)
```
Comment (hopesgit) [inline lib/stat_tracker.rb]: Can you just put an underscore between game/team and "id" here?

**Review by hopesgit — APPROVED** [review]

File: lib/game.rb [hunk 2]
```diff
@@ -45,4 +45,8 @@ def winner_id
     'tie'
   end
 
+  def get_opponent_id(teamid)
+    teamid == @away_team_id ? @home_team_id : @away_team_id
+  end
```
Comment (jakeheft) [inline lib/game.rb]: call this method in statracker, rather than created a new one

File: lib/games_manager.rb [hunk 3]
```diff
@@ -161,14 +161,14 @@ def worst_season(team_id)
...
     @games.find do |game|
-      game.game_id == gameid
+      game.game_id == game_id
     end
   end
 
-  def get_opponent_id(game, teamid)
-    game.away_team_id == teamid ? game.home_team_id : game.away_team_id
+  def get_opponent_id(game_id, teamid)
+    get_game(game_id).get_opponent_id(teamid)
```
Comment (jakeheft) [inline lib/games_manager.rb]: Change to team_id

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -213,8 +189,8 @@ def games_by_team(team_id)
...
+    @games_manager.get_opponent_id(gameid, teamid)
```
Comment (jakeheft) [inline lib/stat_tracker.rb]: add underscore here as well

Comment (dunlapww) [reply lib/stat_tracker.rb]: Yes, thanks! added

File: lib/stat_tracker.rb (hunk 1 above)
Comment (dunlapww) [reply lib/stat_tracker.rb]: added! thanks

**Review by jakeheft — APPROVED** [review]

**Comment by jakeheft** [comment]
Everything looks good after you changed those ids

**Pull request merged**